src/main/java/org/ow2/proactive/catalog/repository/CatalogObjectRepository.java

```diff
@@ -52,9 +52,15 @@
     @Query(value = "SELECT DISTINCT cos.contentType FROM CatalogObjectEntity cos")
     Set<String> findAllContentTypes();
 
+    @Query(value = "SELECT cos.bucket.bucketName, cos.id.name FROM CatalogObjectEntity cos WHERE lower(cos.kind) LIKE lower(concat('%', ?1, '%')) AND lower(cos.contentType) LIKE lower(concat('%', ?2, '%'))")
+    List<Object[]> findCatalogObjectNameReferenceByKindAndContentType(String kind, String contentType);
+
     @Query(value = "SELECT cos.bucket.bucketName, cos.id.name FROM CatalogObjectEntity cos WHERE lower(cos.kind) LIKE lower(concat('%', ?1, '%'))")
     List<Object[]> findCatalogObjectNameReferenceByKind(String kind);
```

Comment:
this method can be removed if you will use just general version with contentType = ""
findCatalogObjectNameReferenceByKindAndContentType